jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java

```diff
@@ -160,34 +159,29 @@ private static Reader getFileReaderFromDir(File file) {
         return result;
     }
 
-    private static String resolveImageTagFromArgs(String imageTagString, Map<String, String> args) {
-        if (imageTagString.startsWith("$")) { // FROM $IMAGE
-            String resolvedVal = resolveArgValueFromStrContainingArgKey(imageTagString, args);
-            if (resolvedVal != null) {
-                return resolvedVal;
-            }
-        } else { // FROM image:$TAG_ARG
-            String[] imageTagArr = imageTagString.split(":");
-            if (imageTagArr.length > 1) {
-                String tag = resolveArgValueFromStrContainingArgKey(imageTagArr[1], args);
-                if (tag != null) {
-                    return imageTagArr[0] + ":" + tag;
-                }
+    static String resolveImageTagFromArgs(String imageTagString, Map<String, String> args) {
+        String resolvedImageString = imageTagString;
+        Set<String> foundArgs = findAllArgs(imageTagString);
+        for (String foundArg : foundArgs) {
+            if (args.containsKey(foundArg)) {
+                resolvedImageString = resolvedImageString.replaceFirst(String.format("\\$\\{*%s\\}*", foundArg),
+                    args.get(foundArg));
             }
         }
-        return imageTagString;
+        return resolvedImageString;
     }
 
-    static String resolveArgValueFromStrContainingArgKey(String argString, Map<String, String> args) {
-        Matcher matcher = argPattern.matcher(argString);
-        if (matcher.matches()) {
-            if (matcher.group(1) != null) {
-                return args.get(matcher.group(1));
-            } else if (matcher.group(2) != null) {
-                return args.get(matcher.group(2));
+    static Set<String> findAllArgs(String imageTagString) {
+        Matcher m = Pattern.compile(ARG_PATTERN_REGEX).matcher(imageTagString);
```

Comment:
The static Pattern instance was removed and now we're instantiating it for each call.